client/my-sites/checkout/cart/cart-messages.tsx

```diff
@@ -105,6 +105,16 @@ function getBlockedPurchaseErrorMessage( {
 	);
 }
 
+function getNoDomainMappingForEmailErrorMessage( {
+	translate,
+}: {
+	translate: ReturnType< typeof useTranslate >;
+} ) {
+	return translate(
+		'We have removed an email product from your cart because you no longer have the domain it depends on.'
```

Comment:
Would have been nice if this said instead:
I'm not sure that the error message is super helpful from the point of view of a new user. What about:

`We have removed Professional Email for the domain <domain.com> from your cart because you no longer have the domain it depends on.`

While it also looks like this error code is unique to the eligibility checks, would it always be in the context of an email product?

Reply:
I am definitely open to improving the error message!
The big gotcha here is that we only have the error code and error message from the server. So in this context, we can't pull out the product name reliably. We can infer that it's an email product, because the domain/email eligibility logic is the only place we return this error, but not what kind of email product.

Reply:
I think we have a way to pass additional parameters in the error returned by our REST API, Igor mentioned that to me some time ago.

Reply:
> I think we have a way to pass additional parameters in the error returned by our REST API

If we know what type of product is generating the error, could we return different error codes for each type of product? (That's the only way I'm aware of.)

More importantly, I wonder if this change could be made to the error message on the server itself? The mechanism of `getMessagePrettifier` is meant to enable displaying HTML messages from the server, since those can't safely be rendered from an API response into React, but in this case there's no HTML in this message.